src/util/time.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * @file time.js
+ */
+const findSegmentForTime = (time, playlist) => {
+
+  if (!playlist.segments || playlist.segments.length === 0) {
+    return;
+  }
+
+  // Assumptions:
+  // - there will always be a segment.duration
+  // - we can start from zero
+  // - segments are in time order
+  // - segment.start and segment.end only come
+  //    from syncController
+
+  let manifestTime = 0;
+
+  for (let i = 0; i < playlist.segments.length; i++) {
+    const segment = playlist.segments[i];
+    const estimatedStart = manifestTime;
+    const estimatedEnd = manifestTime + segment.duration;
+
+    if (segment.start <= time && time <= segment.end) {
+      return {
+        segment,
+        estimatedStart,
+        estimatedEnd,
+        type: 'accurate'
+      };
+    } else if (estimatedStart <= time && time <= estimatedEnd) {
+      return {
+        segment,
+        estimatedStart,
+        estimatedEnd,
+        type: 'estimate'
+      };
+    }
+
+    manifestTime = estimatedEnd;
+  }
+
+  return null;
+};
+
+export const getStreamTime = ({
```

Comment:
Is there a reason we are passing `onsuccess` and `onreject` as callbacks instead of returning a promise?

Reply:
We want this to be compatible with IE and we don't polyfill Promises so we tend not to use them in video.js projects